breathecode/certificate/tests/urls/tests_root.py

```diff
@@ -0,0 +1,114 @@
+    def test_certificate_cohort_user__with_auth_without_permissions(self):
+        """Test /root with auth without permissions"""
+        self.headers(academy=1)
+        self.generate_models(authenticate=True)
+        url = reverse_lazy('certificate:root')
+        response = self.client.post(url, {})
+        json = response.json()
+
+        expected = {
+            'detail': "You (user: 1) don't have this capability: crud_certificate for academy 1",
+            'status_code': 403
+        }
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, 403)
+
+    """
+    🔽🔽🔽 Post method
+    """
+    @patch(GOOGLE_CLOUD_PATH['client'], apply_google_cloud_client_mock())
+    @patch(GOOGLE_CLOUD_PATH['bucket'], apply_google_cloud_bucket_mock())
+    @patch(GOOGLE_CLOUD_PATH['blob'], apply_google_cloud_blob_mock())
+    def test_certificate_re_attemps_with_role_student(self):
+        """Test /root with auth"""
+        """ Good Request """
+        self.headers(academy=1)
+        model = self.generate_models(authenticate=True, cohort=True, user=True,
+            profile_academy=True, capability='crud_certificate', role='STUDENT', 
+            cohort_user=True, syllabus=True, specialty=True, layout_design=True)
+
+        url = reverse_lazy('certificate:root')
+        data = [{
+            'cohort_slug':  model['cohort'].slug,
+            'user_id':  model['user'].id,
+        }]
+        response = self.client.post(url, data, format='json')
+        json = response.json()
+        expected = {
+            'detail': 'The certificates have been scheduled for generation'
+        }
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(self.all_user_specialty_dict(), [])
+
+    @patch(GOOGLE_CLOUD_PATH['client'], apply_google_cloud_client_mock())
+    @patch(GOOGLE_CLOUD_PATH['bucket'], apply_google_cloud_bucket_mock())
+    @patch(GOOGLE_CLOUD_PATH['blob'], apply_google_cloud_blob_mock())
+    def test_certificate_re_attemps_without_cohort_user_and_capability(self):
+        """Test /root with auth"""
+        """ No capability for the request"""
+        self.headers(academy=1)
+        model = self.generate_models(authenticate=True, cohort=True, user=True,
+            profile_academy=True)
+
+        url = reverse_lazy('certificate:root')
+        data = [{
+            'cohort_slug':  model['cohort'].slug,
+            'user_id':  model['user'].id,
+        }]
+        response = self.client.post(url, data, format='json')
+        json = response.json()
+        expected = {
+            'detail': "You (user: 1) don't have this capability: crud_certificate for academy 1",
+            'status_code': 403
+        }
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(self.all_user_specialty_dict(), [])
```

Comment:
We need any test that generate any certificate, for now it not exist

Reply:
Este test lo creo que la issue https://github.com/4GeeksAcademy/About-4Geeks-Academy/issues/1284 que es donde he cambiado las condiciones del POST.

Reply:
We need the test here in this pull request, try to provide de basic data to check that the endpoint work, check it checking you can pass a valid request and in the db it make a certificate with the correct status